### PR TITLE
Split various composite systems out of System.h into their own compilation units. 

### PR DESCRIPTION
**Pull request by david-german-tri**
Migrate the test coverage for the composite systems to googletest.  Replaces #1752.

### CODE REVIEW

**Comment by david-german-tri** [comment]
@jamiesnape Carrying over the conversation from #1752, the Jenkins CI is [failing](https://drake-jenkins.csail.mit.edu/job/experimental/192/compiler=clang,label=linux/consoleFull) on 

`"/usr/local/bin/cmake" "-DCMAKE_BUILD_TYPE:STRING=Release" "-GUnix Makefiles" "/home/ubuntu/workspace/experimental/compiler/clang/label/linux/drake"`

I can reproduce this failure locally, but it's not something that happens in the ordinary course of building Drake with `make`.  Any idea why Jenkins runs this command?  I'm trying to figure out whether the failure is a legit deficiency in the way I've added googletest to the build, or an artifact caused by a brittle CI configuration.

If you don't know offhand, I'd also be grateful for any pointers toward how Jenkins is configured.

File: drake/CMakeLists.txt [hunk 1]
```diff
@@ -143,6 +143,11 @@ endif()
 
 enable_testing()
 include(CTest)
+find_package(GTest REQUIRED)
+include_directories(${GTEST_INCLUDE_DIRS})
+if (GTEST_FOUND)
+  set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -pthread")
```
Comment (RussTedrake) [inline drake/CMakeLists.txt]: you are adding -pthread to every link line with this.  is there a more selective way?  some other libs include it via their pkgconfig

Comment (RussTedrake) [inline drake/CMakeLists.txt]: perhaps adding it to GTEST_BOTH_LIBRARIES?  it also might want to be more platform specific (windows doesn't have pthread...?)

Comment (david-german-tri) [inline drake/CMakeLists.txt]: I think this flag is no longer necessary post 8b11750, so I've removed it in 334f89e.

**Comment by tkoolen** [comment]
Those last two commits were as 'Drake Refactor Bot', just so you're aware :-)

**Comment by david-german-tri** [comment]
Good catch, thanks.  As far as I can tell, there's no way to change the author for the merge, but at least I've fixed my configuration going forward.  Three cheers for git.

**Comment by jamiesnape** [comment]
retest this please

**Comment by david-german-tri** [comment]
Retest in progress (apparently - I didn't actually do anything to kick that off).  Would you mind filling me in on what you're looking for?

**Comment by jamiesnape** [comment]
http://drake002.csail.mit.edu/drake/sphinx/jenkins.html

**Comment by david-german-tri** [comment]
Oh.  Neat.

**Comment by david-german-tri** [comment]
@jamiesnape Setting aside the new breakage on OS X (which I'll look into this afternoon), Jenkins is still running that mysterious CMake command on Linux.  I'd be super grateful for any insight you can offer into why that happens, or at least pointers to the code/config I should read.

**Comment by jamiesnape** [comment]
This is the script that is being run:

https://github.com/billhoffman/drake/blob/build-scripts/scripts/drake_jenkins_ctest.cmake

**Comment by david-german-tri** [comment]
- 76d15d7 fixed the OS X build by configuring an absolute INSTALL_NAME for the dylibs.
- 6f304e5 fixed the Linux CI build by ensuring that GTest is findable

So, I think that were it not for the breakage at head from #1814, this PR would finally be ready to merge.  It's been a long and instructive adventure in cross-platform build...

**Comment by david-german-tri** [comment]
@RussTedrake At long last, this PR is green!

**Pull request merged**